### ISSUE TITLE
runtime: document that MemStats.Lookups is always zero

### ISSUE THREAD

**Issue by zc24** [golang/go#55915]
package main
import "fmt"
import "runtime"
func main () {
    var a = runtime.MemStats{}
    runtime.ReadMemStats(&a)
    fmt.Printf("%+v\n", a.Lookups)
}

**Comment by dmitshur** [comment]
Please include answers to questions from the issue template:

- What did you do?
- What did you expect to see?
- What operating system and processor architecture are you using (`go env`)?

Note that the [release policy](https://go.dev/doc/devel/release#policy) means there will be minor releases for Go 1.19 and 1.18, not older.

**Comment by zc24** [comment]
@dmitshur hello，I want to get the value of runtime.MEMstats.Lookups in the latest version 1.19, but It's always going to be 0. I see that in the 1.11 submission history, someone changed this, and in the end, lookups was discarded, but the reason is not explained clearly.I want to know the reason for the abandonment

**Comment by go101** [comment]
Yes, 1.10 prints 3. But many runtime internal implementation details change from version to version. So this is not a sufficient evidence to prove the field is discarded.

**Comment by zc24** [comment]
@go101 @dmitshur Which method will trigger the increase of Lookups in the latest version 1.19

**Comment by go101** [comment]
It does looks the Lookups field is never used as L-values now.

**Comment by ZekeLu** [comment]
I think this is changed by 41e6abdc61dd23ede4d3509aebf7b5a638f53712. The comment for the next field was changed:

https://github.com/golang/go/blob/41e6abdc61dd23ede4d3509aebf7b5a638f53712/src/runtime/mstats.go#L29

```diff
-   nlookup     uint64 // number of pointer lookups
+   nlookup     uint64 // number of pointer lookups (unused)
```

**Comment by zc24** [comment]
@ZekeLu I looked at the submission information, but I couldn't see the fundamental reason why the lookups count was dropped, perhaps because the statistic itself doesn't mean much.The field is still there for downward compatibility

**Comment by mknyszek** [comment]
What Michael said is correct, the documentation is just out of date. As for why it was set to zero, one reason why might be that this needs to be tracked on a very hot path in the runtime, so it's a cost everyone would pay for.

I'll update the docs.

**Comment by mknyszek** [comment]
This field has basically been deprecated since Go 1.11. I don't know the procedure we should follow for deprecating it. I filed https://github.com/golang/go/issues/60714 to find out.

**Comment by mknyszek** [comment]
I think this is basically blocked on #60714. That will decide the ultimate fate of `MemStats.Lookups`. It doesn't look like this is going to happen for Go 1.21 either (though, it's definitely not urgent).

Closing in favor of #60714.
